agera/src/main/java/com/google/android/agera/BaseObservable.java

```diff
@@ -130,16 +130,17 @@ private void add(@NonNull final Updatable updatable, @NonNull final Handler hand
         if (updatablesAndHandlers[index] == updatable) {
           throw new IllegalStateException("Updatable already added, cannot add.");
         }
-        if (updatablesAndHandlers[index] == null && !added) {
```

Comment:
this removal will fail some duplication tests -- the old code ensures the updatable to add does not appear in the array at a higher index, after using a null slot at a lower index.